### PR TITLE
add off topic filter and auto focus

### PR DESCRIPTION
**Pull request by kristianmitk**

### CODE REVIEW

File: client/src/app/learn-page/learn-page.component.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
 import { RecommenderService } from '../recommender.service';
 import { ActivatedRoute } from '@angular/router';
-import { Subscription } from 'rxjs';
+import {combineLatest, forkJoin, Subscription} from 'rxjs';
```
Comment (bersling) [inline client/src/app/learn-page/learn-page.component.ts:4]: did you install the precommit hooks? looks wrongly formatted.

File: client/src/app/taskbase-ui/tb-chat-input/tb-chat-input.component.ts [hunk 2]
```diff
@@ -7,23 +7,41 @@ import { Subscription } from 'rxjs';
...
+  inputElement: ElementRef | undefined;
+
   subscriptions: Subscription[] = [];
   constructor(private chatService: ChatService) {}
 
   ngOnInit() {
     this.chatService.disabled.subscribe((isDisabled) => {
       this.disabled = isDisabled;
+      if(!isDisabled) {
+        // why is this not always working?
```
Comment (bersling) [inline client/src/app/taskbase-ui/tb-chat-input/tb-chat-input.component.ts:25]: don't know. should i play around with it and try to find out?

File: server/src/controllers/feedback.py [hunk 3]
```diff
@@ -34,52 +34,37 @@ def compute_feedback(request: dict):
...
-#             feedback = feedback + positive_feedback
-#         except Exception as e:
-#             print(e)
-#
-#     return feedback
+
+@router.post("/api/feedback/off-topic-filter")
+def filters(request: dict):
+    # general filter task
+    filter_task_id = "akYulFrAa5n6nUUBiHNyBD"
```
Comment (bersling) [inline server/src/controllers/feedback.py:60]: could make this an env variable

File: client/src/app/learn-page/learn-page.component.ts (hunk 1 above)
Comment (kristianmitk) [reply client/src/app/learn-page/learn-page.component.ts:4]: Of course I did, not. :) Will format

**Pull request merged**